src/permissions.ts

```diff
@@ -1,31 +1,48 @@
-import { PermissionObject } from './types';
+/*
+ * RoleName can have any custom string value depending on application configuration. Please do not check role value directly
+ * and consider using channel capabilities to detect available features
+ */
+export type RoleName = string;
 
-type RequiredPermissionObject = Required<PermissionObject>;
-
-export const Allow = 'Allow';
-export const Deny = 'Deny';
-export const AnyResource = ['*'];
-export const AnyRole = ['*'];
-export const MaxPriority = 999;
-export const MinPriority = 1;
+export type Role = {
+  created_at: string;
+  custom: boolean;
+  name: RoleName;
+  scopes: string[];
+  updated_at: string;
+};
 
-// deprecated permission object class, you should use the new permission system v2 and use permissions
-// defined in BuiltinPermissions to configure your channel types
+export type Permission = {
+  action?: string;
+  condition?: object;
+  custom?: boolean;
+  description?: string;
+  id?: string;
+  level?: string;
+  name?: string;
+  owner?: boolean;
+  same_team?: boolean;
+};
 
-export class Permission {
-  name: RequiredPermissionObject['name'];
-  action: RequiredPermissionObject['action'];
-  owner: RequiredPermissionObject['owner'];
-  priority: RequiredPermissionObject['priority'];
-  resources: RequiredPermissionObject['resources'];
-  roles: RequiredPermissionObject['roles'];
+/*
+ * Policies are entities for permissions v1. They are deprecated and will be removed in the future release
+ */
+export class Policy {
+  action?: 'Deny' | 'Allow';
```

Comment:
There's a check failing related to permissions, is that related to this change? Idk if it's just a matter of renaming in the test if so, but 🤞🏻